Fix completion.web_history.exclude not working when set in config.py

mamg22 opened the pull request:
<!-- Thanks for submitting a pull request! Please pick a descriptive title (not just "issue 12345"). If there is an open issue associated to your PR, please add a line like "Closes #12345" somewhere in the PR description (outside of this comment) -->

The completion is now regenerated when `completion.web_history.exclude` is set or changed in `config.py`.

I tested this on both `config.py` and `autoconfig.yml` with the setting set, unset and modified, found no issues or crashes.

I think this could be used to replace the rebuild mechanism used exclusively by the `autoconfig.yml`, since it rebuilds any time the patterns change in the already loaded config. I'm not really sure since I'm not familiar enough with the codebase, yet. 

Closes #5868

The-Compiler commented on qutebrowser/browser/history.py:
```diff
@@ -156,9 +157,18 @@ def __init__(self, progress, parent=None):
         self.completion = CompletionHistory(parent=self)
         self.metainfo = CompletionMetaInfo(parent=self)
 
+        # Get a string of all patterns
+        patterns = "\n".join([str(pattern) for pattern in
+                              config.cache['completion.web_history.exclude']])
```
While this should probably work, you can get a string out of the config system directly via:

```suggestion
        config.instance.get_str('completion.web_history.exclude')
```

mamg22 replied:
Change applied in commit 93ecd8f

The-Compiler commented on qutebrowser/browser/history.py:
```diff
@@ -80,6 +80,7 @@ class CompletionMetaInfo(sql.SqlTable):
 
     KEYS = {
         'force_rebuild': False,
```
Like you mentioned in your comment, I think you can get rid of `force_rebuild` and the old `_on_config_changed` entirely.

The-Compiler reviewed (changes requested):
Thanks! I added a couple of comments. This will also need a test, check `test_force_rebuild` in `tests/unit/browser/test_history.py` for inspiration.

rcorre reviewed (commented):
LGTM, pending @The-Compiler's suggestions. Thanks!

The-Compiler commented:
@mamg22 Any update on this?

mamg22 commented:
> 
> 
> @mamg22 Any update on this?

Yeah, I think I'm halfway done, just need to write the tests. Was busy with university homework until a few days ago.

The-Compiler commented:
Alright, no worries - take your time and let me know if I can help somewhere!

mamg22 commented:
Thanks, yesterday I was setting up the test environment to check what tests broke by the removal of "force_rebuild", but I can't run any of them because it keeps failing with `Failed: Got logging message on logger misc with level WARNING: YAML load took unusually long`.

I've been trying to make pyyaml use the C extension so it could load it faster, but I can't get it to build or use that, even though I have installed both the C and C++ development libraries for `libyaml`, is there way I can force it to build the C extension when running it in tox?

btw, I'm using `tox -e py36 -- tests/unit/browser/test_history.py` to run the tests.

mamg22 commented:
All changes are ready :).

Applied requested changes, added a test for the completion rebuild on config change and changed tests that depended on "force_rebuild".

The-Compiler commented:
Thanks @mamg22 for the bugfix and @rcorre for the review!

Looks good now - I improved a test slightly and also pushed a change to remove the old `force_rebuild` key to not confuse people who look at the database manually:

- e609f3585 Improve test_modify
- cd0000f72 history: Remove old force_rebuild values
- a9c72a43b Update changelog
